### PR TITLE
Don't copy XUnit runner to output for local builds

### LINKED ISSUE #94183: `dotnet test` is not running tests for .NETFramework targets
### Description

When running `dotnet test` on a test project in dotnet/runtime that targets .NETFramework I expect tests to run, but they do not.  Seems like we have lots of other issues mentioning this, but I couldn't find an active one.

### Reproduction Steps

build clr+libs -rc Release
cd src\libraries\System.Collections.Immutable\tests
dotnet test -f:net462

### Expected behavior

Tests run.
```
    Discovering: System.Collections.Immutable.Tests (app domain = on [no shadow copy], method display = ClassAndMethod,
   method display options = None)
    Discovered:  System.Collections.Immutable.Tests (found 4008 of 4123 test cases)
    Starting:    System.Collections.Immutable.Tests (parallel test collections = on, max threads = 16)
```

### Actual behavior

No tests run:
```
Test run for c:\src\dotnet\runtime\artifacts\bin\System.Collections.Immutable.Tests\Debug\net462\System.Collections.Immutable.Tests.dll (.NETFramework,Version=v4.6.2)
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
No test is available in c:\src\dotnet\runtime\artifacts\bin\System.Collections.Immutable.Tests\Debug\net462\System.Collections.Immutable.Tests.dll. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.
```

### Regression?

I think this is a recent regression.  I believe this used to work as I remember seeing test failures on framework when running with dotnet test.

### Known Workarounds

Directly invoke the `test` target.

> \src\dotnet\runtime\.dotnet\dotnet build /t:test -f:net462

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by ericstj**
Fix #94183 

This fixes it for me, and addresses the problem that @akoeplinger mentioned previously that made us revert.

We should try to throw as many pipelines as possible at this change to see if there are any other special cases where folks are expecting the runner in the output.

### CODE REVIEW

File: eng/testing/xunit/xunit.console.targets [hunk 1]
```diff
@@ -75,8 +75,8 @@
 
   <!-- ResolveAssemblyReferences is the target that populates ReferenceCopyLocalPaths which is what is copied to output directory. -->
   <Target Name="CopyRunnerToOutputDirectory" BeforeTargets="ResolveAssemblyReferences" Condition="'$(TargetsMobile)' != 'true' or '$(BundleXunitRunner)' == 'true'">
-    <ItemGroup>
-      <!-- Copy test runner to output directory -->
+    <ItemGroup Condition="'$(ArchiveTests)' == 'true' or '$(BundleXunitRunner)' == 'true'">
+      <!-- When ArchiveTests or BundleXunitRunner is set, copy xunit.runner.console to output directory -->
```
Comment (ViktorHofer) [inline eng/testing/xunit/xunit.console.targets:79]: nit:

```suggestion
      <!-- When ArchiveTests or BundleXunitRunner is set, copy xunit.console to output directory -->
```

**Review by ViktorHofer — APPROVED** [review]
(Pending extensive validation)

**Comment by ericstj** [comment]
Odd error from ILC @sbomer 
https://dev.azure.com/dnceng-public/public/_build/results?buildId=782237&view=logs&j=ea7d6d51-d18d-5306-47bd-e3b7db04467e&t=6f613bc6-161e-53a2-c035-24718f3c6909&l=11413
```
Unhandled exception: System.AggregateException: One or more errors occurred. (Code generation failed for method '[S.P.CoreLib]Internal.Runtime.CompilerHelpers.ThrowHelpers.ThrowInvalidProgramException(ExceptionStringID)')
     ---> ILCompiler.CodeGenerationFailedException: Code generation failed for method '[S.P.CoreLib]Internal.Runtime.CompilerHelpers.ThrowHelpers.ThrowInvalidProgramException(ExceptionStringID)'
     ---> ILCompiler.NativeAotFatalErrorException: ILC: error IL1013: Error processing 'name'.
     ---> System.NullReferenceException: Object reference not set to an instance of an object.
       at System.Xml.XPath.XNodeNavigator.MoveToFirstChild() + 0x22
       at MS.Internal.Xml.XPath.XPathChildIterator.MoveNext() + 0x19
       at System.Xml.XPath.XPathNodeIterator.Enumerator.MoveNext() + 0xd6
       at ILCompiler.ProcessLinkerXmlBase.ProcessAssemblies(XPathNavigator) + 0x157
       at ILCompiler.ProcessLinkerXmlBase.ProcessXml(Boolean) + 0xfa
```
I don't think it should be caused by this change.  All other NativeAOT legs passed.  Will rerun.

**Comment by ericstj** [comment]
> I don't think it should be caused by this change. All other NativeAOT legs passed. Will rerun.

Rerun fixed this, so I think there must be a transient issue.  This looks a little bit like the JIT issues around tailcall (spurious `NullReferenceException` without explanation).  This is running on Preview7.  cc @AndyAyersMS

**Comment by AndyAyersMS** [comment]
> > I don't think it should be caused by this change. All other NativeAOT legs passed. Will rerun.
> 
> Rerun fixed this, so I think there must be a transient issue. This looks a little bit like the JIT issues around tailcall (spurious `NullReferenceException` without explanation). This is running on Preview7. cc @AndyAyersMS

Seems like it very well could be, yes...

**Comment by ericstj** [comment]
@akoeplinger can you have a look and double check the testing?  I've run all the build definitions I can find that might be relevant.  I want to double check with you that this covers the stuff that broke before.

**Comment by ericstj** [comment]
I scanned through all the failures, nothing looked to me like failure to launch the test runner - just a bunch of "normal" test failures.

**Comment by ericstj** [comment]
> Azure Pipelines could not run because the pipeline triggers exclude this branch/path.

Ok then, I will run it myself: https://dev.azure.com/dnceng-public/public/_build/results?buildId=785913&view=results

**Comment by ericstj** [comment]
OK, I think this should be fixed now.  `/t:Test` was indeed broken because it calls the testrunner script.  I made an update to fix the path to xunit in that script.

Later we should look at pulling xunit out even for the helix case and using an external/parameterized path in that script.  Not looking to make that change just yet.

Here's the new build of enterprise-linux tests: https://dev.azure.com/dnceng-public/public/_build/results?buildId=786949&view=results

File: eng/testing/xunit/xunit.console.targets [hunk 2]
```diff
@@ -12,10 +17,15 @@
     <SetScriptCommands Condition="'$(TargetOS)' != 'windows'" Include="export XUNIT_HIDE_PASSING_OUTPUT_DIAGNOSTICS=1" />
   </ItemGroup>
 
+  <PropertyGroup Condition="'$(BundleXunitRunner)' == 'true'">
+    <XunitConsolePath Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">xunit.console.dll</XunitConsolePath>
+    <XunitConsolePath Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">xunit.console.exe</XunitConsolePath>
+  </PropertyGroup>
```
Comment (ViktorHofer) [inline eng/testing/xunit/xunit.console.targets:23]: Why do we need to reset this property instead of using what's defined in L8-9?

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**